tests/functional/autoprompt/test_prompttoolkit.py

```diff
@@ -146,3 +234,273 @@ def test_handle_args_with_spaces(self):
         self.assertEqual(prompter.input_buffer.document.text,
                          "iam create-role --description 'With spaces' ")
         self.assertEqual(args, original_args)
+
+
+class TestPromptToolkitDocBuffer(BasicPromptToolkitTest):
+    def get_updated_doc_buffer_text(self, original_args):
+        self.prompter.args = original_args
+        self.prompter.pre_run()
+        return self.prompter.doc_buffer.document.text
+
+    def test_doc_buffer_not_shown_on_start_and_not_focusable(self):
+        stubber = ApplicationStubber(self.prompter.create_application())
+        stubber.add_key_assert(None, lambda app: app.show_doc is False)
+        stubber.add_key_assert(
```

Comment:
It would be nice if we can then add helper methods to any of these common types of assertions so we do not have to build up the lambda each time. For example:
```python
def add_current_buffer_assertion(self, expected_buffer_name)
```
And we can then add assertions in these test cases such as:
```python
self.add_current_buffer_assertion('input_buffer')
```